rfcs/0015-pablo-internals-restructure.md

```diff
@@ -0,0 +1,321 @@
+# RFC-0015
+
+The [current implementation of `DualAssetConstantProduct`][dacp] is tightly
+coupled with the pablo pallet, depending on pablo's `Config` trait directly for
+all of it's functionality and directly modifying pablo's storage. This makes it
+very difficult to test, since a full runtime is required to provide the required
+mocks for it to work properly.
+
+## Goals
+
+1. Remove `DualAssetConstantProduct`'s direct dependency on pablo
+2. Make all of the `DualAssetConstantProduct` functionality pure
```

Comment:
We don't have a requirement to make things "pure" in our code by it self, unless it is to handle an actual product/reuse requirement